server/auth/types/basic/routes.ts

```diff
@@ -186,7 +187,9 @@ export class BasicAuthRoutes {
           }
           context.security_plugin.logger.info('The Redirect Path is ' + redirectUrl);
           try {
-            user = await this.securityClient.authenticateWithHeaders(request, {});
+            user = await this.securityClient.authenticateWithHeaders(request, {
+              authorization: ANONYMOUS_AUTH_HEADER,
+            });
```

Comment:
code between lines 214-228 is dead-code because `multi_tenancy` is never set upon receiving auth response from backend.

The dead-code:
https://github.com/opensearch-project/security-dashboards-plugin/blob/3c0c2f981dcf4d0bea26ae25ac8d20e02911f9a2/server/auth/types/basic/routes.ts#L214-L227

this is the reason:

https://github.com/opensearch-project/security-dashboards-plugin/blob/3c0c2f981dcf4d0bea26ae25ac8d20e02911f9a2/server/backend/opensearch_security_client.ts#L98-L104

as we can see that the property `multi_tenancy` for user object is never set, hence the `if` condition is never true.